core/src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java

```diff
@@ -824,71 +780,74 @@ public static String formatPunishmentScreen(
-            .getPersonalizedText()
-            .color(ChatColor.GRAY)); // The sign-off of who performed the punishment
+    // Staff sign-off - who performed the punishment
+    lines.add(TranslatableComponent.of("moderation.screen.signoff", TextColor.GRAY).args(punisher));
 
     // Link to rules for review by player
     if (PGMConfig.Moderation.isRuleLinkVisible()) {
-      Component rules = new PersonalizedText(PGMConfig.Moderation.getRulesLink());
+      Component rules = TextComponent.of(PGMConfig.Moderation.getRulesLink());
 
-      lines.add(Components.blank());
+      lines.add(TextComponent.empty());
       lines.add(
-          new PersonalizedTranslatable("moderation.screen.rulesLink", rules)
-              .getPersonalizedText()
-              .color(ChatColor.GRAY)); // A link to the rules
+          TranslatableComponent.of("moderation.screen.rulesLink", TextColor.GRAY)
+              .args(rules)); // Link to rules
     }
 
     // Configurable last line (for appeal message or etc)
     if (PGMConfig.Moderation.isAppealVisible() && type.equals(PunishmentType.BAN)) {
-      lines.add(Components.blank());
-      lines.add(new PersonalizedText(PGMConfig.Moderation.getAppealMessage()));
+      lines.add(TextComponent.empty());
+      lines.add(TextComponent.of(PGMConfig.Moderation.getAppealMessage()));
     }
 
-    lines.add(Components.blank());
+    lines.add(TextComponent.empty());
     lines.add(footer); // Footer line - END
 
-    return Components.join(new PersonalizedText("\n" + ChatColor.RESET), lines).toLegacyText();
+    return TextTranslations.translateLegacy(
+        TextComponent.join(TextComponent.of("\n" + ChatColor.RESET), lines),
+        null); // TODO add viewer
   }
 
   /*
    * Sends a formatted title and plays a sound warning a user of their actions
    */
   private void sendWarning(MatchPlayer target, String reason) {
-    Component titleWord =
-        new PersonalizedTranslatable("misc.warning")
-            .getPersonalizedText()
-            .color(ChatColor.DARK_RED);
-    Component title = new PersonalizedText(WARN_SYMBOL, titleWord, WARN_SYMBOL);
-    Component subtitle = formatPunishmentReason(reason).color(ChatColor.GOLD);
-
-    target.showTitle(title, subtitle, 5, 200, 10);
+    Component titleWord = TranslatableComponent.of("misc.warning", TextColor.DARK_RED);
+    Component title =
+        TextComponent.builder().append(WARN_SYMBOL).append(titleWord).append(WARN_SYMBOL).build();
+    Component subtitle = formatPunishmentReason(reason).color(TextColor.GOLD);
+
+    // TODO: Upgrade show title to new text
+    tc.oc.pgm.util.component.Component oldTitle =
+        new PersonalizedText(TextTranslations.translateLegacy(title, target.getBukkit()));
+    tc.oc.pgm.util.component.Component oldSubTitle =
+        new PersonalizedText(TextTranslations.translateLegacy(subtitle, target.getBukkit()));
+
+    target.showTitle(oldTitle, oldSubTitle, 5, 200, 10);
```

Comment:
I would modify `Audience` to add `showTitle` and the hotbar with the new components. Make them defaults and use `TextTranslations`